gulpfile.js

```diff
@@ -5,16 +5,37 @@ const electron = require('electron-connect').server.create({
 });
 const tape = require('gulp-tape');
 const tapColorize = require('tap-colorize');
+const ts = require("gulp-typescript");
+const tsProject = ts.createProject("tsconfig.json");
+
+const glob = require('glob');
+const { execSync } = require('child_process');
+
+const baseFilePattern = 'app/+(main|renderer)/**/*';
+const globOptions = { cwd: __dirname };
+const jsFiles = glob.sync(baseFilePattern + '.js', globOptions);
+const tsFiles = glob.sync(baseFilePattern + '.ts', globOptions);
+if (jsFiles.length !== tsFiles.length) {
+	const npx = process.platform === 'win32' ? 'npx.cmd' : 'npx';
```

Comment:
Is this not required `tools/run-dev`? There we seem to be directly using `npx`.

Reply:
We do use for `tools/run-dev` but it's visible; the `run` function takes care of this.